src/main/java/ch/kalunight/zoe/repositories/GameInfoCardRepository.java

```diff
@@ -0,0 +1,208 @@
+
+  private static final String UPDATE_GAME_INFO_CARDS_MESSAGES_WITH_ID =
+      "UPDATE game_info_card SET gamecard_titlemessageid = %d, " +
+          "gamecard_infocardmessageid = %d, " +
+          "gamecard_creationtime = '%s' WHERE gamecard_id = %d";
+  
+  private static final String UPDATE_GAME_INFO_CARD_STATUS_WITH_ID =
+      "UPDATE game_info_card SET gamecard_status = '%s' WHERE gamecard_id = %d";
+
+  private static final String INSERT_INTO_GAME_INFO_CARD = "INSERT INTO game_info_card " +
+      "(gamecard_fk_infochannel, gamecard_fk_currentgame, gamecard_status) VALUES (%d, %d, '%s')";
+
+  private GameInfoCardRepository() {
+    //hide default public constructor
+  }
+  
+  public static void updateGameInfoCardStatusWithId(long gameCardId, GameInfoCardStatus status)
+      throws SQLException {
+    try (Connection conn = RepoRessources.getConnection();
+        Statement query = conn.createStatement();) {
+
+      String finalQuery = String.format(UPDATE_GAME_INFO_CARD_STATUS_WITH_ID, status.toString(), gameCardId);
+      query.executeUpdate(finalQuery);
+    }
+  }
+  
+  public static DTO.GameInfoCard getGameInfoCardsWithCurrentGameId(long guildId, long currentGameId) throws SQLException {
+    ResultSet result = null;
+    try (Connection conn = RepoRessources.getConnection();
+        Statement query = conn.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);) {
+
+      String finalQuery = String.format(SELECT_GAME_INFO_CARD_WITH_CURRENT_GAME_ID_AND_GUILDID,
+          guildId, currentGameId);
+      result = query.executeQuery(finalQuery);
+      int rowCount = result.last() ? result.getRow() : 0;
+      if(rowCount == 0) {
+        return null;
+      }
+
+      return new DTO.GameInfoCard(result);
+    }finally {
+      RepoRessources.closeResultSet(result);
+    }
+  }
+
+  public static void updateGameInfoCardsMessagesWithId(long titleId, long messageId, LocalDateTime creationTime, long gameCardId)
+      throws SQLException {
+    try (Connection conn = RepoRessources.getConnection();
+        Statement query = conn.createStatement();) {
+
+      String finalQuery = String.format(UPDATE_GAME_INFO_CARDS_MESSAGES_WITH_ID, titleId, messageId,
+          DTO.DB_TIME_PATTERN.format(creationTime), gameCardId);
+      query.executeUpdate(finalQuery);
+    }
+  }
+
+
+  public static void createGameCards(long infochannelId, long currentGameId, GameInfoCardStatus status) throws SQLException {
+    try (Connection conn = RepoRessources.getConnection();
+        Statement query = conn.createStatement();) {
+
+      String finalQuery = String.format(INSERT_INTO_GAME_INFO_CARD, infochannelId, currentGameId, status.toString());
+      query.execute(finalQuery);
+    }
+  }
+
+  public static DTO.GameInfoCard getGameInfoCardsWithCurrentGame(long guildId, long currentGameInfoId, long leagueAccountId)
+      throws SQLException {
+    ResultSet result = null;

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet.](https://app.codacy.com/manual/KaluNight/Zoe-Discord-Bot/pullRequest?prid=4739924)